ppx/test/errors.t/run.t

```diff
@@ -0,0 +1,19 @@
+  $ dune build ./prettify.exe
+$ ocamlopt -dsource _build/default/prettify.pp.ml
+  $ dune exec ./prettify.exe -- ok.json
+  {
+    "a": [ "Foo" ],
+    "foo": [ [ "A" ], [ "Foo" ], [ "B", 123 ], [ "C", 234, "hello" ] ],
+    "b": "where are you ?",
+    "c": [ 123, 234, 345 ],
+    "d": [ 123, [ 1.2, 2.3, 2.4 ], "i am here" ]
+  }
+  $ dune exec ./prettify.exe -- tag_as_string.json
+  Fatal error: exception  exception Ppx_deriving_json_runtime.Of_json_error(Json_error "expected [\"A\"] or [\"Foo\"] or [\"B\", _] or [\"C\", _, _] or [\"D\", { _ }]. Json received: \"A\"")
```

Comment:
I think this is 10x better than before, congrats! do you think adding the key would be even better? 

`expected blablalb at key "xxx"` or some other wording

Reply:
Maybe. I think the "proper" way to do it could be to give a loc in the input file, because there won't always be a key (the variant might be in a list for instance). But getting locs is probably impossible (the yojson datastructure does not have them). So I can try and add more context in some places.

Reply:
I'd also replace `or` with `|` to resemble the OCaml compiler's error messages for warning 8.

Reply:
right, there may be cases where the key isn't on the level where you are decoding but still doable. If it gets messy, can always push another PR with it, don't want to block anything here

Reply:
> I'd also replace or with | to resemble the OCaml compiler's error messages for warning 8.

I think `or` is more appropriate here because `|` is not part of the json syntax, and the errors are expressed in json syntax.